engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java

```diff
@@ -229,23 +214,9 @@ private long scanLog(final long snapshotPosition) {
           }
         }
 
-        final UnifiedRecordValue recordValue =
-            recordValues.readRecordValue(newEvent, metadata.getValueType());
-        typedEvent.wrap(newEvent, metadata, recordValue);
-
-        if (MigratedStreamProcessors.isMigrated(typedEvent)
-            && typedEvent.getRecordType() == RecordType.COMMAND) {
-          // store the highest key of a processed command for supporting legacy processors
-          // - initialize the map to store the key of the follow-up record afterward
-          lastGeneratedKeyBySourceCommandPosition.put(currentPosition, -1L);
-        }
-
         final var recordKey = newEvent.getKey();
         // records from other partitions should not influence the key generator of this partition
         if (Protocol.decodePartitionId(recordKey) == zeebeState.getPartitionId()) {
-          lastGeneratedKeyBySourceCommandPosition.computeIfPresent(
-              sourceEventPosition, (position, key) -> Math.max(recordKey, key));
-
           // remember the highest key on the stream to restore the key generator after replay
           highestRecordKey = Math.max(recordKey, highestRecordKey);
         }
```

Comment:
After merging this and https://github.com/camunda-cloud/zeebe/pull/7446 we can completely remove the scan method

Reply:
:thinking: yes :+1: 
We can move the additional logic (e.g. last source event position, highest record key) to the other loop.